cookbook/case_studies/ml_training/spark_horovod/keras_spark_rossmann_estimator.py

```diff
@@ -36,421 +38,608 @@
     Reshape,
 )
 
+CATEGORICAL_COLS = [
+    "Store",
+    "State",
+    "DayOfWeek",
+    "Year",
+    "Month",
+    "Day",
+    "Week",
+    "CompetitionMonthsOpen",
+    "Promo2Weeks",
+    "StoreType",
+    "Assortment",
+    "PromoInterval",
+    "CompetitionOpenSinceYear",
+    "Promo2SinceYear",
+    "Events",
+    "Promo",
+    "StateHoliday",
+    "SchoolHoliday",
+]
+
+CONTINUOUS_COLS = [
+    "CompetitionDistance",
+    "Max_TemperatureC",
+    "Mean_TemperatureC",
+    "Min_TemperatureC",
+    "Max_Humidity",
+    "Mean_Humidity",
+    "Min_Humidity",
+    "Max_Wind_SpeedKm_h",
+    "Mean_Wind_SpeedKm_h",
+    "CloudCover",
+    "trend",
+    "trend_de",
+    "BeforePromo",
+    "AfterPromo",
+    "AfterStateHoliday",
+    "BeforeStateHoliday",
+    "BeforeSchoolHoliday",
+    "AfterSchoolHoliday",
+]
 
-############
-# GET DATA #
-############
-@task(cache=True, cache_version="0.1",)
+
+@dataclass_json
+@dataclass
+class Hyperparameters:
+    batch_size: int = 100
+    sample_rate: float = 0.01
+    learning_rate: float = 0.0001
+    num_proc: int = 2
+    epochs: int = 100
+    local_checkpoint_file: str = "checkpoint.h5"
+    local_submission_csv: str = "submission.csv"
+
+
+@task(
+    cache=True,
+    cache_version="0.1",
+)
 def download_data() -> FlyteDirectory:
+    # create a directory named 'data'
+    print("==============")
+    print("Downloading data")
+    print("==============")
+
     working_dir = flytekit.current_context().working_directory
     data_dir = pathlib.Path(os.path.join(working_dir, "data"))
     data_dir.mkdir(exist_ok=True)
 
-    download_subp = subprocess.run(
+    # download
+    download_subprocess = subprocess.run(
         [
             "curl",
             "https://cdn.discordapp.com/attachments/545481172399030272/886952942903627786/rossmann.tgz",
```

Comment:
i think can we not simply take this as an input? And have this as the default value?

Reply:
this task can then take the file and untar it?

Reply:
I think that particular dataset can be the only one value because our code wouldn't work with a different dataset. So we needn't give the dataset as an input.

Reply:
but still why not make this as the default? i think we need to keep showing inputs are important